pkgs/applications/office/libreoffice/update.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p python3 pup curl jq nix-prefetch nix
+
+set -euo pipefail
+echoerr() { echo "$@" 1>&2; }
+
+fname="$1"
+echoerr got fname $fname
+shift
+
+variant="$1"
+# See comment near version_major variable
+if [[ $variant == fresh ]]; then
+    head_tail=head
+elif [[ $variant == still ]]; then
+    head_tail=tail
+else
+    echoerr got unknown variant $variant
+    exit 3
+fi
+echoerr got variant $variant
+shift
+
+# Not totally needed, but makes it easy to run the update in case tis folder is
+# deleted.
+mkdir -p "$(dirname $fname)/src-$variant"
+cd "$(dirname $fname)/src-$variant"
+
+# The pup command prints both fresh and still versions one after another, and
+# we use either head -1 or tail -1 to get the right version, per the if elif
+# above.
+version_major="$(curl --silent https://www.libreoffice.org/download/download-libreoffice/ |\
+    pup '.dl_version_number text{}' | $head_tail -1)"
+echoerr got from website ${variant}_version $version_major
+baseurl=https://download.documentfoundation.org/libreoffice/src/$version_major
+tarballs=($(curl --silent $baseurl/ |\
+    pup 'table json{}' |\
+    jq --raw-output '.. | .href? | strings' |\
+    grep "$version_major.*.tar.xz$"))
+
+full_version="$(echo ${tarballs[0]} | sed -e 's/^libreoffice-//' -e 's/.tar.xz$//')"
+echoerr full version is $full_version
+echo \"$full_version\" > version.nix
+
+for t in help translations; do
+    nix-prefetch --check-store fetchurl --url $baseurl/libreoffice-$t-$full_version.tar.xz --output nix > $t.nix
+done
+
+# We don't use `nix-prefetch --output nix` here, because we want the know the
+# output path, and we want the tarball to actually exist in the /nix/store, to
+# then extract the `download.lst` file for ../generate-libreoffice-srcs.py.
+main_path_hash=($(nix-prefetch-url --print-path $baseurl/libreoffice-$full_version.tar.xz))
+echo "{" > main.nix
+echo "  sha256 = "\"${main_path_hash[0]}'";' >> main.nix
```

Comment:
it can be `hash` instead.

Reply:
> it can be `hash` instead.

I was thinking about it, but apparently `nix-prefetch-url` prints the sha256 in base 16, and not 64. I'm not sure how to convert between the two, and I wanted to have consistency among all files `main.nix` `help.nix` and `translations.nix`. `nix-prefetch` writes `sha256` and not `hash`, but uses base 64 and I couldn't find a way to change that behavior.

Reply:
Why not to get rid of `nix-prefetch`, then? The main repo looks abandoned, and the precise pull request that would allow `hash` was not accepted more than one year.

https://github.com/msteen/nix-prefetch/pull/43

Reply:
> Why not to get rid of `nix-prefetch`, then? The main repo looks abandoned, and the precise pull request that would allow `hash` was not accepted more than one year.
> 
> [msteen/nix-prefetch#43](https://github.com/msteen/nix-prefetch/pull/43)

You have a point. Still though, `nix-prefetch-url` always uses `sha256` in base 16, so I'll have to use `sha256` key in the generated nix files.

I fixed this in a rebase.